keps/sig-cli/3895-kubectl-delete-interactivity/README.md

```diff
@@ -0,0 +1,923 @@
+
+On the other hand, we need to provide a way to mitigate accidental deletes to users and this KEP
+proposes a new flag, namely interactive(-i) that will ask confirmation to user to proceed deletion.
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+### Goals
+
+- Introduce interactive(-i) flag to be used for asking confirmation to user to proceed actual delete operation or not.
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+- Smart solutions on client side to detect deletion is accidental or not
+- deletion protection mechanism on server side
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+User passes `interactive` flag to kubectl delete command and all the resources planned to be deleted will be 
+shown to the user as preview. Command continues deletion process if user confirms by typing `y`. If user types `n` or
+any other value other than `y`, process will be cancelled and command returns a message with zero exit code.
+
+```sh
+$ kubectl delete --interactive -f test_deployment.yaml 
+You are about to delete the following 2 resource(s):
+Deployment/dockerd
+Deployment/dockerd2
+Do you want to continue? (y/n): n
+deletion is cancelled
+
+$ kubectl delete -i -f test_deployment.yaml
+You are about to delete the following 2 resource(s):
+Deployment/dockerd
+Deployment/dockerd2
+Do you want to continue? (y/n): y
+deployment.apps "dockerd" deleted
+deployment.apps "dockerd2" deleted
+```
+
+For the alpha phase, this flag will be hidden behind `KUBECTL_INTERACTIVE_DELETE` environment variable and
+can only be use when this `KUBECTL_INTERACTIVE_DELETE=true` environment variable is exported.
+
+### Technical Implementation
+
+One of major advantage of this feature is the ability of showing resources as preview before deletion. That means
+we need to traverse all resources twice. First one is for confirmation and the other one is for deletion.
+However, that comes along with a technical limitation that `Visit` function does not support traversing twice
+as stated in [here](https://github.com/kubernetes/kubernetes/blob/6d7a9048b67d81e57923892650a23636f1afba42/staging/src/k8s.io/cli-runtime/pkg/resource/result.go#L91-L94).
+
```

Comment:
Drop this paragraph, you're continuing your thought below and paragraph is misleading.